dali/kernels/imgproc/color_manipulation/brightness_contrast_test.cc

```diff
@@ -0,0 +1,230 @@
+
+TYPED_TEST(BrightnessContrastTest, check_kernel) {
+  BrightnessContrastCPU<typename TypeParam::in, typename TypeParam::out> kernel;
+  check_kernel<decltype(kernel)>();
+}
+
+
+TYPED_TEST(BrightnessContrastTest, SetupTestAndCheckKernel) {
+  BrightnessContrastCPU<typename TypeParam::in, typename TypeParam::out> kernel;
+  constexpr auto ndims = std::remove_reference_t<decltype(*this)>::ndims;
+  KernelContext ctx;
+  InTensorCPU<typename TypeParam::in, ndims> in(this->input_.data(), this->shape_);
+  auto reqs = kernel.Setup(ctx, in, this->brightness_, this->contrast_);
+  auto sh = reqs.output_shapes[0][0];
+  ASSERT_EQ(this->shape_, sh);
+}
+
+
+TYPED_TEST(BrightnessContrastTest, RunTest) {
+  BrightnessContrastCPU<typename TypeParam::in, typename TypeParam::out> kernel;
+  constexpr auto ndims = std::remove_reference_t<decltype(*this)>::ndims;
+  KernelContext ctx;
+  InTensorCPU<typename TypeParam::in, ndims> in(this->input_.data(), this->shape_);
+  auto reqs = kernel.Setup(ctx, in, this->brightness_, this->contrast_);
+  auto out_shape = reqs.output_shapes[0][0];
+  vector<typename TypeParam::out> output;
+  output.resize(dali::volume(out_shape));
+  OutTensorCPU<typename TypeParam::out, ndims> out(output.data(),
+                                                   out_shape.template to_static<ndims>());
+
+  kernel.Run(ctx, out, in, this->brightness_, this->contrast_);
+  for (int i = 0; i < out.num_elements(); i++) {
+    EXPECT_FLOAT_EQ(this->ref_out_tv_.data[i], out.data[i]) << "Failed at idx: " << i;
+  }
+}
+
+
+TYPED_TEST(BrightnessContrastTest, RunTestWithRoi) {
+  BrightnessContrastCPU<typename TypeParam::in, typename TypeParam::out> kernel;
+  constexpr auto ndims = std::remove_reference_t<decltype(*this)>::ndims;
+  KernelContext ctx;
+  InTensorCPU<typename TypeParam::in, ndims> in(this->input_.data(), this->shape_);
+
+  typename decltype(kernel)::Roi roi;
+  brightness_contrast::fill_roi(roi);
+
+  auto reqs = kernel.Setup(ctx, in, this->brightness_, this->contrast_, &roi);
+  auto out_shape = reqs.output_shapes[0][0];
+  vector<typename TypeParam::out> output;
+  output.resize(dali::volume(out_shape));
+  OutTensorCPU<typename TypeParam::out, ndims> out(output.data(),
+                                                   out_shape.template to_static<ndims>());
+
+  kernel.Run(ctx, out, in, this->brightness_, this->contrast_, &roi);
+
+  auto mat = brightness_contrast::to_mat<ndims>(this->ref_output_.data(), roi,
+                                                this->shape_[0], this->shape_[1]);
+  ASSERT_EQ(mat.rows * mat.cols * mat.channels(), out.num_elements())
+                        << "Number of elements doesn't match";
+  auto ptr = reinterpret_cast<typename TypeParam::out *>(mat.data);
+  for (int i = 0; i < out.num_elements(); i++) {
+    EXPECT_FLOAT_EQ(ptr[i], out.data[i]) << "Failed at idx: " << i;
+  }
+}
+
+
+TYPED_TEST(BrightnessContrastTest, roi_shape) {
+  {
+    Box<2, int> box({0, 3});
```

Comment:
I know this _can_ be done this way, but it doesn't mean it _should_.
I'd suggest one of the following:
```suggestion
    Box<2, int> box(ivec2(0), ivec2(3));
```
```suggestion
    Box<2, int> box(0, 3);
```
or
```suggestion
    Box<2, int> box{0, 3};
```